### PR TITLE
Ruby 3 update

### PR DESCRIPTION
**Pull request by jmartin-r7**
Update to utilize Ruby 3.0.2

* Revises Windows `sqlite` install process to provide sqlite in the package matching Linux
* Update `libxml2` and `libxslt`
* Update Ruby to 3.0.2
* Revise `ruby-windows-devkit` install to utilize msys2 env only since previously distributed `devkit` is deprecated
* Update `rubygems`
* Remove unused `ruby-windows-devkit-bash` and `sqlite3-gem`

### CODE REVIEW

**Comment by jmartin-r7** [comment]
MSI build failed will fix.
```
14:30:45         [Builder: sqlite] I | 2021-11-03T19:29:44-05:00 |   LD_RUN_PATH="C:/metasploit-framework/embedded/lib"
14:30:45         [Builder: sqlite] I | 2021-11-03T19:29:44-05:00 | $ C:/metasploit-framework/embedded/msys64/usr/bin/bash.exe -lc ./configure --prefix=C:/metasploit-framework/embedded --disable-readline
14:30:48         [Builder: sqlite] I | 2021-11-03T19:29:47-05:00 | Execute: `C:/metasploit-framework/embedded/msys64/usr/bin/bash.exe -lc ./configure --prefix=C:/metasploit-framework/embedded --disable-readline': 3.0099s
14:30:48         [Builder: sqlite] I | 2021-11-03T19:29:47-05:00 | Build sqlite: 3.0399s
14:30:48 The following shell command exited with status 1:
14:30:48 
14:30:48     $ CFLAGS=-LC:/metasploit-framework/embedded/lib -IC:/metasploit-framework/embedded/include LDFLAGS=-LC:/metasploit-framework/embedded/lib -IC:/metasploit-framework/embedded/include LD_RUN_PATH=C:/metasploit-framework/embedded/lib C:/metasploit-framework/embedded/msys64/usr/bin/bash.exe -lc ./configure --prefix=C:/metasploit-framework/embedded --disable-readline
```

File: docker/ubuntu1204-x86/Dockerfile [hunk 1]
```diff
@@ -41,13 +41,13 @@ RUN mkdir -p /etc/sudoers.d
 RUN echo "jenkins    ALL=NOPASSWD: ALL" > /etc/sudoers.d/jenkins
 RUN chmod 440 /etc/sudoers.d/jenkins
 
-RUN su jenkins -c 'command curl -sSL https://rvm.io/mpapis.asc | gpg --import - && \
-  command curl -sSL https://rvm.io/pkuczynski.asc | gpg --import - && \
-  curl -L -sSL https://get.rvm.io | bash -s stable'
+RUN su jenkins -c 'command curl --insecure -sSL https://rvm.io/mpapis.asc | gpg --import - && \
+  command curl --insecure -sSL https://rvm.io/pkuczynski.asc | gpg --import - && \
+  curl --insecure -L -sSL https://get.rvm.io | bash -s stable'
```
Comment (jmartin-r7) [inline docker/ubuntu1204-x86/Dockerfile:46]: Call using `--insecure` is unfortunate here, other avenues to address this are in progress.  For this PR an image has been built and published for now.

**Comment by jmartin-r7** [comment]
@msjenkins-r7 test this please.

**Comment by adfoster-r7** [comment]
@jmartin-r7 Is this ready for testing? 👀

**Comment by jmartin-r7** [comment]
Yes

**Review by adfoster-r7 — APPROVED** [review]
@sjanusz-r7 tested some workflows with the Ubuntu and Windows setup, and I've additionally verified the RPC layer works still following the previous test steps from https://github.com/rapid7/metasploit-omnibus/pull/127@sjanusz-r7 tested some workflows with the Ubuntu and Windows setup, and I've additionally verified the RPC layer works still following the previous test steps from https://github.com/rapid7/metasploit-omnibus/pull/127

**Pull request merged**